pkg/clusteragent/admission/mutate/autoinstrumentation/rc_provider.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoinstrumentation
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	rcclient "github.com/DataDog/datadog-agent/pkg/config/remote/client"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// remoteConfigProvider consumes tracing configs from RC and delivers them to the patcher
+type remoteConfigProvider struct {
+	client                  *rcclient.Client
+	pollInterval            time.Duration
+	clusterName             string
+	lastProcessedRCRevision int64
+	rcConfigIDs             map[string]struct{}
+
+	cache *instrumentationConfigurationCache
+}
+
+type rcProvider interface {
+	start(stopCh <-chan struct{})
+}
+
+var _ rcProvider = &remoteConfigProvider{}
+
+func newRemoteConfigProvider(
+	client *rcclient.Client,
+	clusterName string,
+	cache *instrumentationConfigurationCache,
+) (*remoteConfigProvider, error) {
+	if client == nil {
+		return nil, errors.New("remote config client not initialized")
+	}
+	return &remoteConfigProvider{
+		client:                  client,
+		clusterName:             clusterName,
+		pollInterval:            10 * time.Second,
+		lastProcessedRCRevision: 0,
+		rcConfigIDs:             make(map[string]struct{}),
+		cache:                   cache,
+	}, nil
+}
+
+func (rcp *remoteConfigProvider) start(stopCh <-chan struct{}) {
+	log.Info("Remote Enablement: starting remote-config provider")
+	rcp.client.Subscribe(state.ProductAPMTracing, rcp.process)
+	rcp.client.Start()
+	ticker := time.NewTicker(rcp.pollInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			log.Info("Remote Enablement: polling configuration from remote-config")
+			rcp.process(rcp.client.GetConfigs(state.ProductAPMTracing), rcp.client.UpdateApplyStatus)
```

Comment:
❓ Question

Why do we need to poll for configs when we already pass `rcp.process` to the `Subscribe` method of the rcp client:

`rcp.client.Subscribe(state.ProductAPMTracing, rcp.process)`